stl/inc/xlocnum

```diff
@@ -251,6 +251,231 @@ __PURE_APPDOMAIN_GLOBAL locale::id numpunct<_Elem>::id;
+             _Seendigit = true, (void) ++_First) {
+            if (_Significant >= _Max_sig_dig) {
+                ++_Power_of_rep_base; // just scale by 10 or 16
+                if (_Idx > 0) {
+                    _Sticky = true;
+                }
+            } else if (_Idx != 0 || _Significant != 0) { // save a significant digit
+                *_Ptr++ = _Src[_Idx];
+                ++_Significant;
+            }
+        }
+    } else {
+        const _Elem _Kseparator = _Punct_fac.thousands_sep();
+        string _Groups(1, _Initial_dec_leading_zero);
+        size_t _Group_index = 0;
+
+        for (; _First != _Last; ++_First) {
+            const size_t _Idx = _STD _Find_elem(_Atoms, *_First);
+            if (_Idx < _Offset_digit_end) { // got a digit, add to group size
+                _Seendigit = true;
+                if (_Significant >= _Max_sig_dig) {
+                    ++_Power_of_rep_base; // just scale by 10 or 16
+                    if (_Idx > 0) {
+                        _Sticky = true;
+                    }
+                } else if (_Idx != 0 || _Significant != 0) { // save a significant digit
+                    *_Ptr++ = _Src[_Idx];
+                    ++_Significant;
+                }
+
+                if (_Groups[_Group_index] != CHAR_MAX) {
+                    ++_Groups[_Group_index];
+                }
+            } else if (_Groups[_Group_index] == '\0' || *_First != _Kseparator) {
+                break; // not a group separator, done
+            } else { // add a new group to _Groups string
+                _Groups.push_back('\0');
+                ++_Group_index;
+            }
+        }
+
+        if (_Group_index != 0) {
+            if (_Groups[_Group_index] > '\0') {
+                ++_Group_index; // add trailing group to group count
+            } else {
+                _Bad = true; // trailing separator, fail
+            }
+        }
+
+        const char* _Grouping_iter      = _Grouping.data();
+        const char* const _Grouping_end = _Grouping.data() + _Grouping.size();
+        char _Current_grouping_count    = '\0';
+        while (!_Bad && _Group_index > 0) {
+            if (_Grouping_iter != _Grouping_end) { // keep the last value when _Grouping is exhausted
+                _Current_grouping_count = *_Grouping_iter; // assign the variable at least once
+                ++_Grouping_iter;
+            }
+
+            --_Group_index;
+            if ((_Current_grouping_count > '\0' && _Current_grouping_count != CHAR_MAX)
+                && ((_Group_index > 0 && _Groups[_Group_index] != _Current_grouping_count)
+                    || (_Group_index == 0 && _Groups[_Group_index] > _Current_grouping_count))) {
+                _Bad = true; // bad group size, fail
+            }
+            // group size okay, advance to next test
+        }
+    }
+
+    if (_Parse_hex && _Seendigit && _Significant == 0) {
+        *_Ptr++ = '0'; // save at least one leading digit for hex

```

Comment:
I would appreciate a further comment that this is true when all of the digits after the `x` and before the decimal point are zero